tests/404_or_greater/expr_match.ml

```diff
@@ -0,0 +1,12 @@
+module M = struct
+  type t = Foo | Bar
+  type r = { i : int; t : t }
+end
+
+let f () =
+  let M.(s) = M.(Foo) in assert (s = M.Foo);
```

Comment:
True.
I'll add 
```OCaml
  let s = M.Bar in
  match s with
  | M.(Foo | Bar) -> assert true
  | _             -> assert false
```

 Do you want me to get rid off all the other stuff? Though we don't rewrite the pattern I can make a hand-wavy argument that it is nice to have since one could still write that type of code.